source/common/profiler/profiler.h

```diff
@@ -66,12 +66,15 @@ class Heap {
 
 /**
  * Default profiler which will be enabled when tcmalloc (not `gperftools`) is used.
+ * This class is not thread-safe.
```

Comment:
can we add ASSERT_IS_MAIN_OR_TEST_THREAD() macro calls to the function impls?

Reply:
Done.